### ISSUE TITLE
Interface Improvements for Mod Support

### ISSUE THREAD

**Issue by Slybo** [qbasicer/qzdl#70]
This is a great app, it's simple and does what it needs to do but could use some improvement. the main interface should have 3 panels instead of 2.
There should be panel on the left that has a list of all saved .zdl profiles. Clicking on one instantly loads it into zdl and shows all of the mods in the "External Files" panel and selects the source ports/Iwad. ("Extra Command Line Arguments" would also be loaded) 
It would make more logical sense to have the panels arranged from left to right: .zdl profiles -> Source Port/IWADS -> External Files. 

Having a default location for loading/saving .zdl files would be another good QOL feature. (It would also be required for the profile panel)

Bonus points for allowing custom icons and/or screenshots for each profile.

**Comment by qs-arno** [comment]
Agreed; being able to export setups is nice (and absolutely essential for allowing mod creators to ship importable files to make setup a breeze), but for day-to-day usage I would much rather have an "internal" list of mod setups than manage *.ZDL files at the OS level.

Seems to me that making ZDL "aware" of the concept of a list of persistent mod profiles could also pave the way for stuff like automatic mod updates in a future release; each mod has a URI in its metadata that contains an update link which ZDL could iterate through to...well, we're getting way out of scope for this issue but you get the idea.  :)

**Comment by goldenmob72** [comment]
It would be nice if you were able to change load (files) orders, like this


https://user-images.githubusercontent.com/40510344/105434008-baf50880-5c28-11eb-8a74-b3ded55a64bf.mp4

**Comment by goldenmob72** [comment]
It would be nice if you were able to change load (files) orders, like this

![105434008-baf50880-5c28-11eb-8a74-b3ded55a64bf](https://user-images.githubusercontent.com/40510344/105434244-29d26180-5c29-11eb-8b75-d30c29765c73.gif)

**Comment by duganchen** [comment]
I was rereading the top post, and it sounds like it would take the model of a music library manager (e.g iTunes) and treat the .zdl files as playlists.

**Comment by qs-arno** [comment]
To me, if we're using a music library manager metaphor, the .zdl files are more like individual songs.  Basically, the idea is to replace having a directory full of .zdl files with a list view inside ZDL (maybe with columns to show metadata?).

**Comment by duganchen** [comment]
These are good ideas, but I'd question whether ZDL is the right place for them. Is the correct approach to redesign ZDL to incorporate them, or to make a new application?

Do have a look at this project, which looks to have been designed for presets from the ground up.

https://github.com/Youda008/DoomRunner
